conjure-core/src/main/java/com/palantir/conjure/parser/types/TypeParser.java

```diff
@@ -90,13 +90,7 @@ public LocalReferenceType parse(ParserState input) throws ParseException {
                 input.rewind();
                 return null;
             }
-            TypeName typeName;
-            try {
-                typeName = TypeName.of(typeReference);
-            } catch (IllegalArgumentException _e) {
-                input.rewind();
-                return null;
```

Comment:
I think the catch here is important in case the input is not expected to be handled by `ForeignReferenceTypeParser` but is handled by `TypeReferenceParser`.
Have you seen https://github.com/palantir/conjure/pull/767/files#r576677144